tests/cxx/overload_expr-d2.h

```diff
@@ -25,15 +25,23 @@ void TplFn3() {
   return OverloadedFn(T{});
 }
 
+template <typename T>
+void TplFn4() {
+  // IWYU: OverloadedFn2 is...*overload_expr-i4.h
```

Comment:
Can you add a comment here to tell the story of why this depends on `OverloadedFn2(int)` and/or `OverloadedFn2(char)`?

Reply:
It could be argued that this expression should not depend on any known `OverloadedFn2` overload, because the type used for overload resolution is a template argument, unbound at this point.

It would be nice to cover the whys and whats of why that isn't happening here too.

Reply:
> It could be argued that this expression should not depend on any known OverloadedFn2 overload, because the type used for overload resolution is a template argument, unbound at this point.

Reporting of the possible overloads of an unbound (dependent) function call was exactly the original purpose of `VisitOverloadExpr`. The suggestion should be the same as on the master branch and as on the old IWYU releases, I suspect (AFK now). I suggested to remove the old logic in #1529.